Content.Server/GameTicking/Rules/TraitorRuleSystem.cs

```diff
@@ -329,4 +326,51 @@ private void OnRoundEndText(RoundEndTextAppendEvent ev)
         }
         ev.AddLine(result);
     }
+
+    public List<Traitor.TraitorRole> GetOtherTraitorsAliveAndConnected(Mind.Mind ourMind)
+    {
+        var traitors = Traitors;
+        List<Traitor.TraitorRole> removeList = new();
+
+        foreach (var traitor in traitors) // we remove if...
```

Comment:
It would be less duplicate code if instead of the foreach loop and `removeList` you did something like this:
```csharp
return Traitors
    .Where(t => t.Mind is not null) // no mind
    .Where(t => t.Mind.OwnedEntity is not null) // no entity
    ...
```

Reply:
I heard using linq was bad but I could write it that way if we confirm linq is not bad

Reply:
It is bad, but there's no better way to do it I don't think. You could do a single if with a massive multi-line condition, but that's not exactly ideal either.

Reply:
uhhh i'm not really a linq expert but it looks like it leads to nullability CBT
![image](https://user-images.githubusercontent.com/60792108/199380923-5342476f-f070-4ef5-8932-64eb34f1cba0.png)
![image](https://user-images.githubusercontent.com/60792108/199380960-e7d5325a-e4d8-4334-9480-b2186222d861.png)

Reply:
casting as entity uid, .Value, ?, making them part of the same statement all don't seem to fix it

Reply:
Yeah, C# is not very smart about nullability but you can use `!` to tell it that something can't be null.

Reply:
> I heard using linq was bad but I could write it that way if we confirm linq is not bad

For hot paths, yeah its bad. It shouldn't be used for things like physics, power, atmos, etc. Or even really any system update method or events that get called regularly.

But for stuff like this that gets run only once or rarely, its fine and can actually make for pretty nice and easily readable code.